relay-server/src/services/buffer/common.rs

```diff
@@ -3,31 +3,154 @@ use relay_base_schema::project::ProjectKey;
 use crate::Envelope;
 
 /// Struct that represents two project keys.
-#[derive(Debug, Clone, Copy, Eq, Hash, Ord, PartialOrd, PartialEq)]
+#[derive(Debug, Clone, Copy)]
 pub struct ProjectKeyPair {
-    pub own_key: ProjectKey,
-    pub sampling_key: ProjectKey,
+    own_key: ProjectKey,
+    sampling_key: Option<ProjectKey>,
```

Comment:
I decided that I would prefer to encode in the pair the lack of sampling key and rather expose different methods for making it behave like previously, where they were both set and the fallback of `sampling_key` was `own_key`.